tests/base/parameterframe/test_parameterframe.py

```diff
@@ -217,6 +217,23 @@ def test_update_values_edits_frames_values(self):
         )
         assert frame.age.source != "a test"
 
+    def test_update_from_dict(self):
+        frame = BasicFrame.from_dict(FRAME_DATA)
+
+        frame.update_from_dict(
+            {"height": {"value": 160.4, "unit": "m", "source": "a test"}}
+        )
+
+        assert frame.height.value == 160.4
+        assert frame.height.source == "a test"
+        assert (
+            frame.age.value
+            == pint.Quantity(FRAME_DATA["age"]["value"], FRAME_DATA["age"]["unit"])
+            .to("s")
+            .magnitude
+        )
+        assert frame.age.source != "a test"
+
```

Comment:
Did you mean to assert on `age`? I believe we're updating `height` here, not `age`

Reply:
yeah I've added an age test too.